templates/python/custom-copilot-rag-azure-ai-search/src/custom_say_command.py

```diff
@@ -0,0 +1,106 @@
+import json
+from botbuilder.core import MessageFactory, TurnContext
+from botframework.connector import Channels
+from teams.ai.citations import AIEntity, ClientCitation, Appearance
+from teams.utils import snippet
+from teams.utils.citations import format_citations_response, get_used_citations
+from typing import Dict, Any
+
+async def say_command(context: TurnContext, state: Any, data: Dict[str, Any], feedback_loop_enabled: bool = False) -> str:
+    """
+    Executes the say command, sending a message to the user with the AI's response.
+
+    Args:
+        context (TurnContext): The turn context.
+        state (Any): The turn state.
+        data (Dict[str, Any]): The data containing the AI's response.
+        feedback_loop_enabled (bool): Whether the feedback loop is enabled.
+
+    Returns:
+        str: An empty string.
+    """
+    # Early return if there's no valid response data to process
+    if not data or not data.response or not data.response.content:
+        return ""
+
+    # Determine if we're in Teams channel for special formatting
+    is_teams_channel = context.activity.channel_id == Channels.ms_teams
+    content = ""
+    result = None
+
+    try:
+        # Attempt to parse the response as JSON
+        result = json.loads(data.response.content)
+    except json.JSONDecodeError as error:
+        # If not valid JSON, fallback to sending raw text with AI metadata
+        print(f"Response is not valid JSON, sending the raw text. error: {error}")
+        message = MessageFactory.text(data.response.content)
+        # Add schema.org metadata to identify this as AI-generated content
+        message.entities = [{
+            "type": "https://schema.org/Message",
+            "@type": "Message",
+            "@context": "https://schema.org",
```

Comment:
minor: does Teams-AI/BotFramework also provides a type for `entities`/`channel_data` property?